### PR TITLE
[FLINK-26851][metrics] Properly model reporter options as ConfigOptions

### PR DESCRIPTION
**Pull request by zentol**
The first commit extends the SuffixOption to include a key prefix that is also written into the generated html file.
This allows us to use the same file for documenting the fully-qualified option (prefix + suffix) or just the suffix option (e.g., as a list of parameters that a reporter accepts).

The second commits consolidates generic reporter options as ConfigOptions in the MetricOptions. The previously used ConfigConstants have been deprecated.

### CODE REVIEW

File: docs/content/docs/deployment/metric_reporters.md [hunk 1]
```diff
@@ -35,18 +35,15 @@ For more information about Flink's metric system go to the [metric system docume
...
-- `metrics.reporter.<name>.scope.delimiter`: The delimiter to use for the identifier (default value use `metrics.scope.delimiter`) for the reporter named `<name>`.
-- `metrics.reporter.<name>.scope.variables.excludes`: (optional) A semi-colon (;) separate list of variables that should be ignored by tag-based reporters (e.g., Prometheus, InfluxDB). 
-- `metrics.reporter.<name>.scope.variables.additional`: (optional) A comma separated map of variables and their values, which are separated by a colon (:). These mappings are added to the variable map by tag-based reporters (e.g. Prometheux, InfluxDB). 
-- `metrics.reporters`: (optional) A comma-separated include list of reporter names. By default all configured reporters will be used.
+Below is a list of parameters that are generally applicable to all reporters.
+Reporters may additionally offer implementation-specific parameters, which are documented in the respective reporter's section.
+
+{{< include_reporter_config "layouts/shortcodes/generated/metric_reporters_section.html" >}}
+
+All properties are configured by setting `metrics.reporter.<reporter_name>.<property>` in the configuration.
```
Comment (metaswirl) [inline docs/content/docs/deployment/metric_reporters.md:43]:
```suggestion
Below is a list of parameters that are generally applicable to all reporters. All properties are configured by setting `metrics.reporter.<reporter_name>.<property>` in the configuration. Reporters may additionally offer implementation-specific parameters, which are documented in the respective reporter's section. 

{{< include_reporter_config "layouts/shortcodes/generated/metric_reporters_section.html" >}}
```

File: docs/content.zh/docs/deployment/metric_reporters.md [hunk 2]
```diff
@@ -35,18 +35,15 @@ For more information about Flink's metric system go to the [metric system docume
...
-- `metrics.reporter.<name>.scope.delimiter`: The delimiter to use for the identifier (default value use `metrics.scope.delimiter`) for the reporter named `<name>`.
-- `metrics.reporter.<name>.scope.variables.excludes`: (optional) A semi-colon (;) separate list of variables that should be ignored by tag-based reporters (e.g., Prometheus, InfluxDB). 
-- `metrics.reporters`: (optional) A comma-separated include list of reporter names. By default all configured reporters will be used.
-- `metrics.reporter.<name>.scope.variables.additional`: (optional) A comma separated map of variables and their values, which are separated by a colon (:). These mappings are added to the variable map by tag-based reporters (e.g. Prometheux, InfluxDB).
+Below is a list of parameters that are generally applicable to all reporters.
+Reporters may additionally offer implementation-specific parameters, which are documented in the respective reporter's section.
+
+{{< include_reporter_config "layouts/shortcodes/generated/metric_reporters_section.html" >}}
+
+All properties are configured by setting `metrics.reporter.<reporter_name>.<property>` in the configuration.
```
Comment (metaswirl) [inline docs/content.zh/docs/deployment/metric_reporters.md:43]:
```suggestion
Below is a list of parameters that are generally applicable to all reporters. All properties are configured by setting `metrics.reporter.<reporter_name>.<property>` in the configuration. Reporters may additionally offer implementation-specific parameters, which are documented in the respective reporter's section. 

{{< include_reporter_config "layouts/shortcodes/generated/metric_reporters_section.html" >}}
```

File: docs/layouts/shortcodes/generated/metric_configuration.html [hunk 3]
```diff
@@ -62,12 +62,36 @@
...
         </tr>
+        <tr>
+            <td><h5>metrics.reporter.&lt;name&gt;.scope.delimiter</h5></td>
+            <td style="word-wrap: break-word;">"."</td>
+            <td>String</td>
+            <td>The delimiter used to assemble the metric identifier for the reporter named &lt;name&gt;.</td>
+        </tr>
+        <tr>
+            <td><h5>metrics.reporter.&lt;name&gt;.scope.variables.additional</h5></td>
+            <td style="word-wrap: break-word;"></td>
```
Comment (metaswirl) [inline docs/layouts/shortcodes/generated/metric_configuration.html:85]:
```suggestion
            <td style="word-wrap: break-word;">(none)</td>
```

**Review by metaswirl — APPROVED** [review]
Looks good. Few nitpick comments, PR should be fine either way.

**Comment by metaswirl** [comment]
Btw, @zentol the failure seems to be related to https://issues.apache.org/jira/browse/FLINK-26721

**Comment by zentol** [comment]
@flinkbot run azure

File: docs/layouts/shortcodes/generated/metric_configuration.html (hunk 3 above)
Comment (zentol) [reply docs/layouts/shortcodes/generated/metric_configuration.html:85]: This file is generated; changing the documented empty default for map options is outside the scope of this issue.

**Pull request merged**